server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/S3WithMinIOGenerationAwareBlobIdTest.java

```diff
@@ -0,0 +1,222 @@
+    private S3BlobStoreDAO s3BlobStoreDAO;
+    private BlobId.Factory blobIdFactory;
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        blobIdFactory = new MinIOGenerationAwareBlobId.Factory(clock, GenerationAwareBlobId.Configuration.DEFAULT, new PlainBlobId.Factory());
+        testee = createBlobStore(blobIdFactory);
+
+        // Why? https://github.com/apache/james-project/pull/1981#issuecomment-2380396460
+        createBucket(testee.getDefaultBucketName().asString());
+    }
+
+    @AfterEach
+    void tearDown() {
+        s3BlobStoreDAO.deleteAllBuckets().block();
+        s3ClientFactory.close();
+    }
+
+    @Override
+    public BlobStore testee() {
+        return testee;
+    }
+
+    @Override
+    public BlobId.Factory blobIdFactory() {
+        return blobIdFactory;
+    }
+
+    private void createBucket(String bucketName) throws Exception {
+        s3ClientFactory.get().createBucket(builder -> builder.bucket(bucketName))
+            .get();
+    }
+
+    public BlobStore createBlobStore(BlobId.Factory blobIdFactory) {
+        AwsS3AuthConfiguration awsS3AuthConfiguration = minoExtension.minioDocker().getAwsS3AuthConfiguration();
+
+        S3BlobStoreConfiguration s3Configuration = S3BlobStoreConfiguration.builder()
+            .authConfiguration(awsS3AuthConfiguration)
+            .region(DockerAwsS3Container.REGION)
+            .uploadRetrySpec(Optional.of(Retry.backoff(3, java.time.Duration.ofSeconds(1))
+                .filter(UPLOAD_RETRY_EXCEPTION_PREDICATE)))
+            .build();
+
+        s3ClientFactory = new S3ClientFactory(s3Configuration, new RecordingMetricFactory(), new NoopGaugeRegistry());
+        s3BlobStoreDAO = new S3BlobStoreDAO(s3ClientFactory, s3Configuration, blobIdFactory, S3RequestOption.DEFAULT);
+
+        return BlobStoreFactory.builder()
+            .blobStoreDAO(s3BlobStoreDAO)
+            .blobIdFactory(blobIdFactory)
+            .defaultBucketName()
+            .deduplication();
+    }
+
+    @ParameterizedTest
+    @MethodSource("storagePolicies")
+    void saveShouldReturnBlobIdOfString(BlobStore.StoragePolicy storagePolicy) {
+        // Given: A BlobStore and its default bucket
+        BlobStore store = testee();
+        BucketName defaultBucketName = store.getDefaultBucketName();
+
+        // When: Saving a blob
+        BlobId blobId = Mono.from(store.save(defaultBucketName, "toto", storagePolicy)).block();
+        String blobIdString = blobId.asString();
+
+        // Then: BlobId string and parsed BlobId should match expectations
+        assertThat(blobIdString).isEqualTo("1/628/M/f/e/m/XjFVhqwZi9eYtmKc5JA9CJlHbVdBqfMuLlIbamY=");
+        assertThat(blobId).isEqualTo(blobIdFactory().parse(blobIdString));
+    }
+
+    @Test
+    @Override
+    @Disabled("S3minio return `Connection: close` in header response, https://github.com/apache/james-project/pull/1981#issuecomment-2380396460")
+    public void deleteBucketConcurrentlyShouldNotFail() {
+    }
```

Comment:
Delete

Reply:
The test will fail if we do that, or do we not need `BlobStoreContract` in S3WithMinIOGenerationAwareBlobIdTest?